### DISCUSSION TITLE
Nested Layouts ? Hmmm

### DISCUSSION THREAD

**Discussion by bloor** [nuxt/nuxt#16229] [Questions]
I have my main layout (header, body[content] , footer)
And then I`m trying to make a sub-layout for the account-user pages, that has a sidebar-menu
![image](https://user-images.githubusercontent.com/1420653/170740651-3af94ef9-5f38-497c-9339-2a9896ed20c9.png)

```
user/account
user/settings
user/billing
etc
```

How I implemented my nested Layout:
![image](https://user-images.githubusercontent.com/1420653/170740569-1a10a496-7619-46f7-9950-f25948293e53.png)

The issue I face is:
Whenever I switch between a `default-layout path/page` .. to a `user-layout path/page` .. the whole page is re-rendered.
The SPA fluidity breaks for some reason. **I'm considering using a Component instead of the `user-layout`.**

So am I doing something wrong when doing nested layouts ?
Nuxt v3.0.0-rc.3 (latest)

Here's the LIVE DEMO:
https://d1map.netlify.app/
```
test-user: antonebogdan@gmail.com
pass: asdasd123
```
When you commute between `features` , `platforms`, `about` everything is fluid, when you go to `account` it breaks.

**Comment by Aksoom-Hussain** [comment]
I would recommend looking at the link
 https://inertiajs.com/pages#persistent-layouts
 
 Here for example, maybe you have an audio player on a podcast website that you want to continue playing as users navigate the site. Or, you may want to maintain the scroll position in your navigation between page visits. In these situations, using persistent layouts is a better choice.
```
layout: (h, page) => h(Layout, () => child),
```

**Reply by bloor** [reply]
Well it's a start. But I would rather wanna use Nuxt3 out of the box solutions (if any) first.
I don't wanna add more bloatware than necessary.  As I said, I currently found a workaround, it's just I`m not happy with the solution and was hoping there's a "good practice" way to do it.

**Reply by ThinkinDragon** [reply]
i am too looking for solution for now i manage through app.vue to nest the main layout and use the custom layout to load the persistence Layout

**Comment by Aksoom-Hussain** [comment]
https://github.com/nuxt/framework/issues/8950

https://nuxt.com/docs/guide/directory-structure/pages#nested-routes

found solution by using the parent page with child by insert the component inside pages/parent.vue
it will be layout -> parent page -> child page
